extensions/applicationinsights-analytics-js/src/JavaScriptSDK/Interfaces/IAnalyticsConfig.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { IExceptionConfig } from "@microsoft/applicationinsights-core-js";
+
+/**
+ * Configuration interface specifically for AnalyticsPlugin
+ * This interface defines only the configuration properties that the Analytics plugin uses.
+ */
+export interface IAnalyticsConfig {
+    /**
+     * A session is logged if the user is inactive for this amount of time in milliseconds.
+     * @default 1800000 (30 minutes)
+     */
+    sessionRenewalMs?: number;
+
+    /**
+     * A session is logged if it has continued for this amount of time in milliseconds.
+     * @default 86400000 (24 hours)
+     */
+    sessionExpirationMs?: number;
+
+    /**
+     * If true, exceptions are not autocollected.
+     * @default false
+     */
+    disableExceptionTracking?: boolean;
+
+    /**
+     * If true, on a pageview, the previous instrumented page's view time is tracked and sent as telemetry and a new timer is started for the current pageview.
+     * @default false
+     */
+    autoTrackPageVisitTime?: boolean;
+
+    /**
+     * If true, default behavior of trackPageView is changed to record end of page view duration interval when trackPageView is called.
+     * @default false
+     */
+    overridePageViewDuration?: boolean;
+
+    /**
+     * Define whether to track unhandled promise rejections and report as JS errors.
+     * @default false
+     */
+    enableUnhandledPromiseRejectionTracking?: boolean;
+
+    /**
+     * Internal flag to track if unhandled promise instrumentation is already set up.
+     * @default false
+     */
+    autoUnhandledPromiseInstrumented?: boolean;
```

Comment:
For configurations that specifically identify that it is used as an internal flag add the following typedoc tags
- @internal Internal use only
- @ignore INTERNAL ONLY